test/CMakeLists.txt

```diff
@@ -283,9 +283,43 @@ make_test(SRC nodevice.cpp CXX11)
 if(OpenCL_FOUND)
   make_test(SRC ocl_ext_context.cpp
             LIBRARIES OpenCL::OpenCL
-            BACKENDS "opencl")
+            BACKENDS "opencl"
+            CXX11)
 endif()
 
+if(CUDA_FOUND)
+  foreach(backend ${enabled_backends})
+    set(cuda_test_backends "cuda" "unified")
+    if(${backend} IN_LIST cuda_test_backends)
+      set(target test_cuda_${backend})
+      cuda_add_executable(${target} cuda.cu)
+      target_include_directories(${target} PRIVATE
+        ${ArrayFire_SOURCE_DIR}/extern/half/include
+        ${CMAKE_SOURCE_DIR}
+        ${CMAKE_CURRENT_SOURCE_DIR})
+      if(${backend} STREQUAL "unified")
+        target_link_libraries(${target}
+          ArrayFire::af)
+      else()
+        target_link_libraries(${target}
+          ArrayFire::af${backend})
+      endif()
+      target_link_libraries(${target}
+        arrayfire_test
+        mmio
+        gtest)
+
+      set_target_properties(${target}
+        PROPERTIES
+        FOLDER "Tests"
+        OUTPUT_NAME "cuda_${backend}")
+
+      add_test(NAME ${target} COMMAND ${target})
+    endif()
+  endforeach()
+endif()
```

Comment:
What am I missing here ? It feels almost identical to `make_test` macro.

Reply:
`cuda_add_executable(${target} cuda.cu)`

Reply:
Too much redundancy for single line difference :(